File: day12_part1_2015.dart

import 'dart:convert';
import 'dart:io';

int sumNumbers(dynamic obj) {
  if (obj is int) {
    return obj;
  } else if (obj is List) {
    return obj.map((e) => sumNumbers(e)).reduce((a, b) => a + b);
  } else if (obj is Map) {
    return obj.values.map((e) => sumNumbers(e)).reduce((a, b) => a + b);
  } else {
    return 0;
  }
}

void main() {
  File file = File('input.txt');
  String input = file.readAsStringSync();
  dynamic data = json.decode(input);
  
  int result = sumNumbers(data);
  print(result);
}

File: training_data/reflection-tuning/day12_part1_2015.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
)

func sumNumbers(data interface{}) float64 {
	var sum float64

	switch v := data.(type) {
	case float64:
		return v
	case []interface{}:
		for _, item := range v {
			sum += sumNumbers(item)
		}
	case map[string]interface{}:
		for _, value := range v {
			sum += sumNumbers(value)
		}
	}

	return sum
}

func main() {
	// Read input file
	content, err := ioutil.ReadFile("input.txt")
	if err != nil {
		fmt.Println("Error reading file:", err)
		return
	}

	// Parse JSON
	var data interface{}
	err = json.Unmarshal(content, &data)
	if err != nil {
		fmt.Println("Error parsing JSON:", err)
		return
	}

	// Calculate sum
	result := sumNumbers(data)

	// Print result
	fmt.Printf("Sum of all numbers: %.0f\n", result)
}
